admin/deployments.go

```diff
@@ -569,6 +558,55 @@ func (s *Service) NewDeploymentAnnotations(org *database.Organization, proj *dat
 	}
 }
 
+func (s *Service) TriggerRuntimeReloadForProject(ctx context.Context, proj *database.Project, environment string) error {
+	projectID := proj.ID
+	ds, err := s.DB.FindDeploymentsForProject(ctx, projectID, environment, "")
+	if err != nil {
+		return err
+	}
+
+	grp, ctx := errgroup.WithContext(ctx)
+	grp.SetLimit(8)
+	for _, d := range ds {
+		isPrimary := proj.PrimaryDeploymentID != nil && *proj.PrimaryDeploymentID == d.ID
+
+		if d.Status != database.DeploymentStatusRunning {
+			if isPrimary {
+				return fmt.Errorf("cannot trigger runtime reload for deployment %q because it is not running", d.ID)
+			}
+			// for non-primary deployments, skip
+			continue
+		}
+		grp.Go(func() error {
+			// Connect to the runtime
+			rt, err := s.OpenRuntimeClient(d)
+			if err != nil {
+				if isPrimary {
+					return err
+				}
+				// for non-primary deployments, log and continue
+				s.Logger.Info("failed to trigger runtime reload", zap.String("deployment_id", d.ID), zap.Error(err), observability.ZapCtx(ctx))
+				return nil
+			}
+			defer rt.Close()
+
+			// Call ReloadConfig
+			_, err = rt.ReloadConfig(ctx, &runtimev1.ReloadConfigRequest{
+				InstanceId: d.RuntimeInstanceID,
+			})
+			if err != nil {
+				if isPrimary {
+					return err
+				}
```

Comment:
Returning the `err` will cause cancellation of other config reload calls on healthy runtimes. So maybe don't return the primary's err from the goroutine, but instead save it in a variable and return it after all goroutines have completed?